ipatests/test_integration/test_ntp_options.py

```diff
@@ -0,0 +1,226 @@
+        cmd = self.client.run_command(['cat', paths.CHRONY_CONF])
+        assert ntp_pool in cmd.stdout_text
+        assert ntp_server in cmd.stdout_text
+        tasks.uninstall_master(self.replica)
+
+        self.cleanup()
+
+    def test_server_and_client_install_with_mixed_options(self):
+        """
+        test to verify that ipa-server and ipa-client install with
+         --ntp-server and -N options would fail
+        """
+        ntp_server = "1.pool.ntp.org"
+        exp_str = ("error: --ntp-server cannot be used"
+                   " together with --no-ntp")
+        exp_pool_str = ("error: --ntp-pool cannot be used"
+                        " together with --no-ntp")
+
+        args1 = ['ipa-server-install', '-N', '--ntp-server=%s' % ntp_server]
+        server_install = self.master.run_command(args1, raiseonerr=False)
+        assert server_install.returncode == 2
+        assert exp_str in server_install.stderr_text
+
+        args2 = ['ipa-client-install', '-N', '--ntp-server=%s' % ntp_server]
+        client_install = self.client.run_command(args2, raiseonerr=False)
+        assert client_install.returncode == 2
+        assert exp_str in client_install.stderr_text
+
+        args3 = ['ipa-client-install', '-N',
+                 '--ntp-pool=%s' % ntp_server.lstrip('1.')]
+        client_install = self.client.run_command(args3, raiseonerr=False)
+        assert client_install.returncode == 2
+        assert exp_pool_str in client_install.stderr_text
+
+    def test_two_step_replica_install_using_ntp_options(self):
+        """
+        test to verify that replica promotion with ntp  --ntp-server,
+         --ntp-pool and -N or --no-ntp option would fail
+        """
+        ntp_server = "1.pool.ntp.org"
+        ntp_pool = "pool.ntp.org"
+        exp_str = "NTP configuration cannot be updated during promotion"
+
+        tasks.install_master(self.master, setup_dns=False)
+        tasks.install_client(self.master, self.replica)
+
+        try:
+            replica_install = self.replica.run_command(
+                ['ipa-replica-install', '-N'], raiseonerr=False)
+            assert replica_install.returncode == 1
+            assert exp_str in replica_install.stderr_text
+
+            replica_install = self.replica.run_command(
+                ['ipa-replica-install', '--ntp-server=%s' % ntp_server],
+                raiseonerr=False)
+            assert replica_install.returncode == 1
+            assert exp_str in replica_install.stderr_text
+
+            replica_install = self.replica.run_command(
+                ['ipa-replica-install', '--ntp-pool=%s' % ntp_pool],
+                raiseonerr=False)
+            assert replica_install.returncode == 1
+            assert exp_str in replica_install.stderr_text
+
+        finally:
+            tasks.uninstall_master(self.replica)
+            self.cleanup()
+
+    def test_two_step_replica_install_without_ntp_options(self):
```

Comment:
Please rename method to:
```suggestion
    def test_replica_promotion_with_ntp(self):
```